src/main.rs

```diff
@@ -71,9 +106,54 @@ async fn main() -> Result<()> {
         warn!("INSECURE: The security of Keylime is NOT linked to a hardware root of trust.");
         warn!("INSECURE: Only use Keylime in this mode for testing or debugging purposes.");
     }
+
+    info!("Starting server...");
+
+    // Gather EK and AK key values and certs
+    let (ek_handle, ek_cert, ek_tpm2b_pub) =
+        tpm::create_ek(&mut ctx, Some(AsymmetricAlgorithm::Rsa))?;
+
+    let (ak_handle, ak_name, ak_tpm2b_pub) =
+        tpm::create_ak(&mut ctx, ek_handle)?;
+
+    // Gather configs
     let cloudagent_ip = config_get("cloud_agent", "cloudagent_ip")?;
     let cloudagent_port = config_get("cloud_agent", "cloudagent_port")?;
-    info!("Starting server...");
+    let registrar_ip = config_get("registrar", "registrar_ip")?;
+    let registrar_port = config_get("registrar", "registrar_port")?;
+    let agent_uuid_config = config_get("cloud_agent", "agent_uuid")?;
+    let agent_uuid = get_uuid(&agent_uuid_config);
+
+    {
+        // Request keyblob material
+        let keyblob = registrar_agent::do_register_agent(
+            &registrar_ip,
+            &registrar_port,
+            &agent_uuid,
+            &ek_tpm2b_pub,
+            &ek_cert,
+            &ak_tpm2b_pub,
+        )
+        .await?;
+        let key = tpm::activate_credential(
+            &mut ctx, keyblob, ak_handle, ek_handle,
+        )?;
+        let mackey = base64::encode(key.value());
+        let mackey = PKey::hmac(&mackey.as_bytes())?;
+        let mut signer = Signer::new(MessageDigest::sha384(), &mackey)?;
+        signer.update(agent_uuid.as_bytes());
+        let auth_tag = signer.sign_to_vec()?;
+        let auth_tag = hex::encode(&auth_tag);
+
+        registrar_agent::do_activate_agent(
+            &registrar_ip,
+            &registrar_port,
+            &agent_uuid,
+            &auth_tag,
+        )
+        .await?;
+    }
```

Comment:
This is the closing bracket: after we activate, we no longer need `keyblob`.

Reply:
Yeah, I did notice it created its own scope that lasted until this line, I just wasn't sure why.